### DISCUSSION TITLE
How to [Insert | Update | Delete]Many on a piece or page type ?

### DISCUSSION THREAD

**Discussion by ezzle** [apostrophecms/apostrophe#4046] [Q&A]
Hi,

Is there a way to do
`... self.updateMany ...`

Best regards,

Tshitshi

**Comment by BoDonkey** [comment]
Sure. The Apostrophe API tries to reduce the overhead of making requests to the database, but updateMany is a valid MongoDB method. You can read about direct DB access here: https://v3.docs.apostrophecms.org/guide/database-access.html#working-with-collections

**Comment by boutell** [comment]
Just to build on what Bob said here, direct MongoDB access can be your
friend in this situation. Just take care to consider aposLocale and the
question of permissions (should this user even be allowed to manipulate
these documents). Which may or may not be a concern depending on your
project.

One way to address permissions is to do .find(req, { ...
}).permission('edit').toArray() when fetching the documents in the first
place, which will only return documents the user is permitted to edit. Then
you have a set of document ids you know it is OK to pass to updateMany.

On Wed, Feb 15, 2023 at 5:31 AM Robert Means ***@***.***>
wrote:

> Sure. The Apostrophe API tries to reduce the overhead of making requests
> to the database, but updateMany is a valid MongoDB method. You can read
> about direct DB access here:
> https://v3.docs.apostrophecms.org/guide/database-access.html#working-with-collections
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/apostrophecms/apostrophe/discussions/4046#discussioncomment-4981807>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAH27NBLR7N6T6SFL6X6HTWXSWA3ANCNFSM6AAAAAAU4OYDSQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: <apostrophecms/apostrophe/repo-discussions/4046/comments/4981807@
> github.com>
>


-- 

THOMAS BOUTELL | CHIEF TECHNOLOGY OFFICER
APOSTROPHECMS | apostrophecms.com | he/him/his

**Reply by ezzle** [reply]
Thank you Tom and Bob.

OK. But What about relationship and relationshipReverse fields ?

**Comment by boutell** [comment]
Perhaps you’d better share your use case in more detail.

On Wed, Feb 15, 2023 at 5:39 PM Tshitshi Kia Ntoni ***@***.***>
wrote:

> Thank you Tom and Bob.
>
> OK. But What about relationship and relationshipReverse fields ?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/apostrophecms/apostrophe/discussions/4046#discussioncomment-4988651>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAH27IK2SUOTLCPDVLPEATWXVLJHANCNFSM6AAAAAAU4OYDSQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>
-- 

THOMAS BOUTELL | CHIEF TECHNOLOGY OFFICER
APOSTROPHECMS | apostrophecms.com | he/him/his
